File: sdk/python/obt_browser/profile_api.py
"""Profile CRUD operations — create, list, update, delete browser profiles."""

from typing import Any, Optional

from ._http import BaseApi


class ProfileApi(BaseApi):
    def list(self) -> list:
        """Return all profiles."""
        return self._get("/api/profiles")

    def create(
        self,
        name: str,
        headless: Optional[bool] = None,
        fingerprint_options: Optional[dict] = None,
        proxy: Optional[dict] = None,
    ) -> dict:
        """Create a new browser profile.

        fingerprint_options keys: os, browser, device, locale
        proxy keys: type (http/https/socks4/socks5), host, port, username, password
        """
        body: dict = {"name": name}
        if headless is not None:
            body["headless"] = headless
        if fingerprint_options:
            body["fingerprintOptions"] = fingerprint_options
        if proxy:
            body["proxy"] = proxy
        return self._post("/api/profiles", body)

    def update(
        self,
        profile_id: str,
        name: Optional[str] = None,
        headless: Optional[bool] = None,
    ) -> dict:
        """Update profile name and/or headless flag."""
        body: dict = {}
        if name is not None:
            body["name"] = name
        if headless is not None:
            body["headless"] = headless
        return self._put(f"/api/profiles/{profile_id}", body)

    def delete(self, profile_id: str) -> None:
        """Permanently delete a profile."""
        self._delete(f"/api/profiles/{profile_id}")

File: sdk/csharp/Api/ProfileApi.cs
using System.Text.Json.Nodes;

namespace ObtBrowser.Api;

/// <summary>Profile CRUD — create, list, update, delete browser profiles.</summary>
public class ProfileApi : BaseApi
{
    public ProfileApi(HttpClient http) : base(http) { }

    /// <summary>List all profiles.</summary>
    public Task<JsonNode?> ListAsync() =>
        GetAsync("/api/profiles");

    /// <summary>
    /// Create a new browser profile.
    /// fingerprintOptions keys: os, browser, device, locale (all optional).
    /// proxy keys: type, host, port, username, password.
    /// </summary>
    public Task<JsonNode?> CreateAsync(
        string name,
        bool? headless = null,
        object? fingerprintOptions = null,
        object? proxy = null)
    {
        var body = new Dictionary<string, object?> { ["name"] = name };
        if (headless.HasValue)          body["headless"] = headless.Value;
        if (fingerprintOptions is not null) body["fingerprintOptions"] = fingerprintOptions;
        if (proxy is not null)          body["proxy"] = proxy;
        return PostAsync("/api/profiles", body);
    }

    /// <summary>Update profile name and/or headless flag. Omit to leave unchanged.</summary>
    public Task<JsonNode?> UpdateAsync(string profileId, string? name = null, bool? headless = null)
    {
        var body = new Dictionary<string, object?>();
        if (name is not null)   body["name"] = name;
        if (headless.HasValue)  body["headless"] = headless.Value;
        return PutAsync($"/api/profiles/{profileId}", body);
    }

    /// <summary>Permanently delete a profile.</summary>
    public Task DeleteAsync(string profileId) =>
        HttpDeleteAsync($"/api/profiles/{profileId}");
}
